lua/barbar/bbye.lua

```diff
@@ -181,6 +119,8 @@ local bbye = {}
 --- @param mods? string|{[string]: any} the modifiers to the command (e.g. `'verbose'`)
 --- @return nil
 function bbye.delete(action, force, buffer, mods)
+  local state = require('barbar.state')
```

Comment:
Didn't want the `get_focus_on_close` code in bbye, but couldn't solve the circular dependency. This is the only non-top level `require` hack I've added. Keeping bbye simple is more important.

Reply:
Removed the hack, wasn't necessary after the refactor. State is now leaf enough that bbye can import it without circular dependency issue. The big issue was the state/render interdependency that is now solved with the event listener pattern.

```
                bbye ->
api -> render --------> state ------------> config/animate/constants/utils
              --------> layout ----------->
```

I don't think splitting `bbye` is a good idea at the moment, it's small enough.

Sidenote but I don't love using new features unless they're needed, e.g. this example. If the old way works, let's keep using the old way until we feel like increasing our minimum supported neovim version. Code is always a liability, the less of it we have, the happier I am.

```lua
local enew = vim.api.nvim_cmd and
  --- The `:enew` command
  --- @param force boolean
  function(force) vim.cmd.enew {bang = force} end or
  --- The `:enew` command
  --- @param force boolean
  function(force) command("enew" .. (force and '!' or '')) end
```

Reply:
> Sidenote but I don't love using new features unless they're needed, e.g. this example.

Will (won't?) do. With this, we should also reconsider what minimum version we want to support in #394.

Reply:
Yeah not sure how often people update their neovim. Probably supporting whatever version debian stable has would be a good conservative target.